### ISSUE TITLE
always_require_non_null_named_parameters false positive

### ISSUE THREAD

**Issue by eernstg** [dart-lang/linter#2294]
**Describe the issue**

always_require_non_null_named_parameters seems to report a false positive for a `required` parameter.

**To reproduce**

Consider the following library:

```dart
class C {
  C({required int? i}): assert(i != null);
}
```

`dartanalyzer` 2.11.0-189.0.dev emits the following diagnostic for this example:

```
  lint • Specify `@required` on named parameters without defaults. • n031.dart:2:20 • always_require_non_null_named_parameters
```

If we follow that advice we get a parameter of the form `@required required <type> <name>`, which seems pointless.

The declaration could have used `int` rather than `int?`, and the assert could be motivated by running with `--no-sound-null-safety`, so the code as such isn't necessarily wrong.

**Expected behavior**

I would expect the lint to _not_ flag a parameter which is already `required`.

**Comment by srawlins** [comment]
I think this is a duplicate of https://github.com/dart-lang/linter/issues/2236.

**Comment by eernstg** [comment]
Right, closing.
